llvm/test/CodeGen/AIE/aie2/end-to-end/Mul2D.ll

```diff
@@ -85,17 +85,17 @@ define void @mul2d(ptr noalias %in_ptr0, ptr noalias %in_ptr1, ptr noalias %out_
 ; CHECK-NEXT:    nop
 ; CHECK-NEXT:    vmov wh6, wl0
 ; CHECK-NEXT:    vmov wh2, wl0
-; CHECK-NEXT:    add r1, r1, #-1; vmul cm0, x6, x2, r0
-; CHECK-NEXT:    jnz r1, #.LBB0_2; vmul cm1, x2, x4, r0
+; CHECK-NEXT:    vmul cm0, x6, x2, r0
+; CHECK-NEXT:    vmul cm1, x2, x4, r0
+; CHECK-NEXT:    add r1, r1, #-1
+; CHECK-NEXT:    jnz r1, #.LBB0_2
 ; CHECK-NEXT:    nop // Delay Slot 5
-; CHECK-NEXT:    nop // Delay Slot 4
-; CHECK-NEXT:    nop // Delay Slot 3
-; CHECK-NEXT:    vst.srs.d8.s32 cm0, s0, [p2], #32 // Delay Slot 2
-; CHECK-NEXT:    vst.srs.d8.s32 cm1, s0, [p2], #32 // Delay Slot 1
+; CHECK-NEXT:    vst.srs.d8.s32 cm0, s0, [p2], #32 // Delay Slot 4
+; CHECK-NEXT:    vst.srs.d8.s32 cm1, s0, [p2], #32 // Delay Slot 3
+; CHECK-NEXT:    nop // Delay Slot 2
+; CHECK-NEXT:    nop // Delay Slot 1
```

Comment:
It seems that the loop-aware scheduling stopped working, we should not have NOPs here. Is this because the vst.srs have a bank conflict with the vlda of the next iteration?

Reply:
If it's not too complicated, I'd be happy if you could add `addrspace` info to some pointers in the IR below.

Reply:
Updated as suggested. The test expected result are not same as before the introduction of AS